.github/scripts/reset-brew.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Copyright (c) 2019 Kevin R Croft <krcroft@gmail.com>
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# This script removes all existing brew packages, and then
+# re-installs some important basic packages (listed below).
```

Comment:
Yes, if we don't care about speeding up rhe job using the cache. 

If we do want to use the cache, then two factors come into play: the smaller the stuff we're caching, the faster it can be restored, and second the combined stuff we're caching needs to compress to less than 400MB. 

Brew fails on both fronts because its carry something like 3.7GB worth in other packages; so we unload those and then have a clean slate and add just what we need.   This is one time even though, and once the cache is created then it will be used for both of our jobs going forward (until they day we want to regenerate a new cache.. which is something we could do monthy). 

You will see a key: value in all the jobs that use the cache. Incrementing its version will invalidate and regenerate a new cache.